src/main/java/edu/yale/its/tp/cas/client/filter/CASFilter.java

```diff
@@ -289,17 +296,35 @@
     public void init(FilterConfig config) throws ServletException
     {
         System.out.println("Initializing CASFilter");
-        casLogin = config.getInitParameter(LOGIN_INIT_PARAM);
-        casValidate = config.getInitParameter(VALIDATE_INIT_PARAM);
-        casServiceUrl = config.getInitParameter(SERVICE_INIT_PARAM);
-        String casAuthorizedProxy = config.getInitParameter(AUTHORIZED_PROXY_INIT_PARAM);
-        casRenew = Boolean.valueOf(config.getInitParameter(RENEW_INIT_PARAM)).booleanValue();
-        casServerName = config.getInitParameter(SERVERNAME_INIT_PARAM);
-        casProxyCallbackUrl = config.getInitParameter(PROXY_CALLBACK_INIT_PARAM);
-        casLogoutCallbackUrl = config.getInitParameter(LOGOUT_CALLBACK_INIT_PARAM);
-        wrapRequest = Boolean.valueOf(config.getInitParameter(WRAP_REQUESTS_INIT_PARAM)).booleanValue();
-        casGateway = Boolean.valueOf(config.getInitParameter(GATEWAY_INIT_PARAM)).booleanValue();
-        remoteUserAttrib = config.getInitParameter(REMOTE_USER_ATTRIB_INIT_PARAM);
+
+        initLogoutList();
+
+        casLogin = Configuration.getParameter(config, LOGIN_INIT_PARAM);
+        casValidate = Configuration.getParameter(config, VALIDATE_INIT_PARAM);
+
+        String casServerUrlPrefix = Configuration.getParameter(config, CAS_SERVER_URL_PREFIX);
+        if (casServerUrlPrefix != null)
+        {
+            if (casLogin == null)
+            {
+                casLogin = casServerUrlPrefix + "/login";
+            }
+            if (casValidate == null)
+            {
+                casValidate = casServerUrlPrefix + "/serviceValidate";
+            }
+        }
+
+        casServiceUrl = Configuration.getParameter(config, SERVICE_INIT_PARAM);
+        String casAuthorizedProxy = Configuration.getParameter(config, AUTHORIZED_PROXY_INIT_PARAM);
+        casRenew = Boolean.valueOf(Configuration.getParameter(config, RENEW_INIT_PARAM)).booleanValue();
+        casServerName = Configuration.getParameter(config, SERVERNAME_INIT_PARAM);
+        casProxyCallbackUrl = Configuration.getParameter(config, PROXY_CALLBACK_INIT_PARAM);
+        casLogoutCallbackUrl = Configuration.getParameter(config, LOGOUT_CALLBACK_INIT_PARAM);
+        wrapRequest = Boolean.valueOf(Configuration.getParameter(config, WRAP_REQUESTS_INIT_PARAM)).booleanValue();

```

Comment:
Boolean value unboxing is unnecessary (Java 5 and after).

`Boolean.valueOf(Configuration.getParameter(config, <param name>)).booleanValue();`

can be simplified to:

`Boolean.valueOf(Configuration.getParameter(config, <param name>));`